radix-engine-derive/src/lib.rs

```diff
@@ -12,32 +15,89 @@ use proc_macro::TokenStream;
 ///     array: Vec<A>
 /// }
 /// ```
-#[proc_macro_derive(Describe, attributes(skip))]
-pub fn describe(input: TokenStream) -> TokenStream {
-    describe::handle_describe(proc_macro2::TokenStream::from(input))
+#[proc_macro_derive(LegacyDescribe, attributes(skip))]
+pub fn legacy_describe(input: TokenStream) -> TokenStream {
+    legacy_describe::handle_describe(proc_macro2::TokenStream::from(input))
+        .unwrap_or_else(|err| err.to_compile_error())
+        .into()
+}
+
+/// Derives code for encoding a struct or enum with Scrypto value model.
+///
+/// # Example
+///
+/// ```ignore
+/// use scrypto::prelude::*;
+///
+/// #[derive(ScryptoEncode)]
+/// pub struct MyStruct {
+///     pub field_1: u32,
+///     pub field_2: String,
+/// }
+/// ```
+#[proc_macro_derive(ScryptoEncode, attributes(sbor))]
+pub fn encode(input: TokenStream) -> TokenStream {
+    encode::handle_encode(proc_macro2::TokenStream::from(input))
         .unwrap_or_else(|err| err.to_compile_error())
         .into()
 }
 
-/// Attribute that derives code to encode, decode and/or describe the struct or enum, using Scrypto data and schema model.
+/// Derives code for decoding a struct or enum with Scrypto value model.
 ///
 /// # Example
 ///
 /// ```ignore
 /// use scrypto::prelude::*;
 ///
-/// #[scrypto(Encode, Decode, Categorize, Describe, NonFungibleData)]
+/// #[derive(ScryptoDecode)]
 /// pub struct MyStruct {
 ///     pub field_1: u32,
 ///     pub field_2: String,
 /// }
 /// ```
-#[proc_macro_attribute]
-pub fn scrypto(attr: TokenStream, item: TokenStream) -> TokenStream {
-    scrypto::handle_scrypto(
-        proc_macro2::TokenStream::from(attr),
-        proc_macro2::TokenStream::from(item),
-    )
-    .unwrap_or_else(|err| err.to_compile_error())
-    .into()
+#[proc_macro_derive(ScryptoDecode, attributes(sbor))]
+pub fn decode(input: TokenStream) -> TokenStream {
+    decode::handle_decode(proc_macro2::TokenStream::from(input))
+        .unwrap_or_else(|err| err.to_compile_error())
+        .into()
+}
+
+/// Derives code for categorizing a struct or enum with Scrypto value model.
+///
+/// # Example
+///
+/// ```ignore
+/// use scrypto::prelude::*;
+///
+/// #[derive(ScryptoCategorize)]
+/// pub struct MyStruct {
+///     pub field_1: u32,
+///     pub field_2: String,
+/// }
+/// ```
+#[proc_macro_derive(ScryptoCategorize, attributes(sbor))]
+pub fn categorize(input: TokenStream) -> TokenStream {
+    categorize::handle_categorize(proc_macro2::TokenStream::from(input))
+        .unwrap_or_else(|err| err.to_compile_error())
+        .into()
+}
+
+/// Derives code for describing a struct or enum with Scrypto schema.
+///
+/// # Example
+///
+/// ```ignore
+/// use scrypto::prelude::*;
+///
+/// #[derive(ScryptoDescribe)]
+/// pub struct MyStruct {
+///     pub field_1: u32,
+///     pub field_2: String,
+/// }
+/// ```
+#[proc_macro_derive(ScryptoDescribe, attributes(sbor))]
```

Comment:
Did you decide not to add `ScryptoSbor` here in the end?

Reply:
I'd also be interested if ScryptoSbor could include both ScryptoDescribe and LegacyDescribe for now, just to check everything compiles okay? :)

Reply:
I will add in another PR, as I considered these "combo" derives as new feature.

Reply:
Ahh sorry, missed that part of the description :)